pkg/controller/endpointslice/reconciler.go

```diff
@@ -287,7 +289,10 @@ func (r *reconciler) reconcileByPortMapping(
 			}
 		}
 
-		// If an endpoint was updated or removed, mark for update or delete
+		// generate the slice labels and check if parent labels have changed
+		labels, svcLabelsUpdated := setEndpointSliceLabels(existingSlice, service)
+
+		// If an endpoint or label was updated or removed, mark for update or delete
```

Comment:
This feels like it's tied to the `if` statement directly below it while it's actually referring to both the `if` and `else if` statements. Maybe move the comment around label changes to the `else if`.